com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineDesiredStateAchievingActionsStep.java

```diff
@@ -47,23 +50,33 @@ protected StepPhase executeStep(ExecutionWrapper context) {
 
     private StepPhase attemptToExecuteStep(ExecutionWrapper execution) {
         CloudApplication app = StepsUtil.getApp(execution.getContext());
-        ApplicationStartupState currentState = computeCurrentState(execution, app);
+        CloudControllerClient client = execution.getControllerClient();
+        CloudApplication cloudApplication = client.getApplication(app.getName());
```

Comment:
Renamed. We need to retrieve it because the application returned from StepsUtil.getApp(...) only contains the application name. All of the other fields are null.